drivers/flash/flash_stm32.c

```diff
@@ -141,6 +144,9 @@ int flash_stm32_wait_flash_idle(const struct device *dev)
 	}
 #if defined(CONFIG_SOC_SERIES_STM32G0X)
 	while ((FLASH_STM32_REGS(dev)->SR & FLASH_SR_BSY1)) {
+#elif defined(CONFIG_SOC_SERIES_STM32H7X)
+	while ((FLASH_STM32_REGS(dev)->SR1 & (FLASH_SR_BSY | FLASH_SR_QW))
+	       || (FLASH_STM32_REGS(dev)->SR2 & (FLASH_SR_BSY | FLASH_SR_QW))) {
```

Comment:
Only checking the QW bit should be sufficient.
Though, you should check that _both_ banks are not busy (but only under #if DUAL_BANK).

Reply:
I'm "unresolving" this until I can see it has been actually resolved.